hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -149,14 +151,14 @@ public void burn(
 	) {
 		validateTrue( type == TokenType.NON_FUNGIBLE_UNIQUE, FAIL_INVALID,
 				"Non fungible burn can be invoked only on Non fungible tokens!");
-		validateFalse(serialNumbers.isEmpty(), FAIL_INVALID,
+		validateFalse(serialNumbers.isEmpty(), INVALID_TOKEN_BURN_METADATA,
 				"Non fungible burn cannot be invoked with no serial numbers");
 		for (final long serialNum : serialNumbers) {
 			ownershipTracker.add(id, OwnershipTracker.forRemoving(id, serialNum));
 			removedUniqueTokens.add(new UniqueToken(id, serialNum, treasury.getId()));
 		}
 		treasury.setOwnedNfts(treasury.getOwnedNfts() - serialNumbers.size());
-		changeSupply(treasuryRelationship, -serialNumbers.size(), FAIL_INVALID);
+		changeSupply(treasuryRelationship, -serialNumbers.size(), INVALID_TOKEN_BURN_METADATA);
```

Comment:
If we throw here it must be because we have a bug. Lets stay with `FAIL_INVALID`